### PR TITLE
Create bigger targets for Header Nav submenu items

### PR DESCRIPTION
**Pull request by mahoneycm**
## Description

Modeled some of the suggestions in #2836 and #3033 which give subnav menu items a larger targeting area and a visual queue that it is selected, similar to the mobile version.

The desired effect was accomplished by setting the submenu item anchor tag to `display: block;` and moving the padding from the parent tag to the anchor itself.

I also set the hover state background color to `primary` from `primary-darker` to give a highlighted effect.

## Additional information

Current Nav menus

image
image

Suggested on #2836 
image

After this fix

image
image
_The padding being added to the anchor tags does make the mega menu larger compared to before_

Fixes https://github.com/uswds/uswds/issues/2836

### CODE REVIEW

**Comment by mahoneycm** [comment]
@mejiaj

**Comment by thisisdano** [comment]
I like the block anchors, but let's remove the highlighting for now. Perhaps it's something we can enable with a setting, but I don't just want to introduce it out of nowhere.

**Comment by thisisdano** [comment]
Let's also look at reducing the padding a bit, feels a bit too generous as-is.

File: src/stylesheets/components/_navigation.scss [hunk 1]
```diff
@@ -283,19 +282,15 @@ $nav-link-arrow-icon-size: 2;
...
-      & + * {
-        margin-top: units(1.5);
-      }
-
       a {
         color: color("white");
-        padding: 0;
         line-height: line-height($theme-navigation-font-family, 3);
+        display: block;
+        padding: units(2);
```
Comment (mejiaj) [inline src/stylesheets/components/_navigation.scss:289]:
```suggestion
        padding: units(1);
```

File: src/stylesheets/components/_navigation.scss [hunk 2]
```diff
@@ -283,19 +282,15 @@ $nav-link-arrow-icon-size: 2;
...
 
         &:hover {
-          background-color: transparent;
+          background-color: color(primary);
```
Comment (mejiaj) [inline src/stylesheets/components/_navigation.scss:292]: Let's remove this for now.

**Review by mejiaj — COMMENTED** [review]
Added some suggestions based on @thisisdano's comments.

**Comment by mahoneycm** [comment]
Requested changes made and pushed! @mejiaj

**Review by mejiaj — COMMENTED** [review]
We might want to update the megamenu link offset value in `_header.scss`. A value of `units('neg-05')` seems to fix it.

Noticed the following regression when focus is on megamenu links.

**Basic with mega-menu**
image
image

**Extended header with mega-menu**
image

**Comment by mahoneycm** [comment]
> We might want to update the megamenu link offset value in `_header.scss`. A value of `units('neg-05')` seems to fix it.
> 
> Noticed the following regression when focus is on megamenu links.
> 
> **Basic with mega-menu** image image
> 
> **Extended header with mega-menu** image

Fixed in commit [4cea191](https://github.com/uswds/uswds/pull/4427/commits/4cea191cdb2f6ccc9d56e790540a6f37e6d1d5df). Note that I had to add this to `_navigation.scss` to get it to work. Not `_header.scss`

**Review by mejiaj — APPROVED** [review]

**Comment by thisisdano** [comment]
@mahoneycm Looks like this just needs some merge conflict resolution and we're good to go

**Pull request closed without merging**